myapp/playlist.js

```diff
@@ -0,0 +1,67 @@
+var songJSON=JSON.parse(playlistsJSON);
+var l=songJSON.length;
+
+var playlist=[];
+var songs=[];
+
+
+
+function Playlist(data){
```

Comment:
Separate out the Playlist definition into it's own file. Add setter/getters for all fields. Put the logic to construct the list of Song objects for each playlist with the Playlist "class"